### ISSUE TITLE
Create automatic docker build when pusing to github

### ISSUE THREAD

**Issue by inspiraluna** [Doichain/dapp#132]
- create build on docker hub when Github gets updated (pushed)
- use Doichain dApp version as automatic docker tag version

**Comment by jay-dee7** [comment]
@inspiraluna is the bounty open to public? Or is it just for internal references?

**Comment by inspiraluna** [comment]
@jay-dee7 thank you for your interaction. Actually I personally started doing it the other day but it still has issues. If you'd like to help, I'd be more then happy for some support from your side as you seem to be an expert on this side. I'd offer a bounty of 50 DOI here more if its a bigger problem but I think its just a little problem as always.  So far it triggers a build on docker hub. But when building the image it cannot find certain files: See: https://github.com/Doichain/dapp/tree/master/contrib/docker/dapp

**Comment by jay-dee7** [comment]
Sure @inspiraluna. I'll give it a shot today evening 😁

**Comment by jay-dee7** [comment]
@inspiraluna one quick question, do you want to build the image using github actions and then push it to docker hub or dockerhub-github integration from hub.docker.com?

**Comment by inspiraluna** [comment]
Afaik hub.docoker.com installs a webhook on github in the end.  And as somebody pushs to github it informs hub.docker.com to build a certain Dockerfile. So I guess its the dockerhub-github integration from hub.docker.com which asks  the github profile. You could make a fork from our doichain/dapp repository and reference it on your own hub.docker.com.  It should build the above mentioned Dockerfile inside here: https://github.com/Doichain/dapp/tree/master/contrib/docker/dapp

**Comment by jay-dee7** [comment]
i cant see any dockerfile inside this dir: https://github.com/Doichain/dapp/tree/master/contrib/docker/dapp
the readme in there refers to a dockerfile in here: https://github.com/Doichain/docker.git

**Comment by inspiraluna** [comment]
can you switch to branch 0.0.9 ?

**Comment by jay-dee7** [comment]
yeah sure @inspiraluna. found it

**Comment by jay-dee7** [comment]
@inspiraluna am i doing something wrong or are the build times really high? it's been 40+ minutes on docker hub and it's still building

**Comment by inspiraluna** [comment]
yes, I think the build time is quite high and also the resulting image is in my opinion far to big. Thinking on another bounty to improve that. What do you think about this?

**Comment by jay-dee7** [comment]
Another bounty sounds great to me. There's a tool named `dive` i guess. We can inspect docker layers and try to optimize it

**Comment by inspiraluna** [comment]
I created https://github.com/Doichain/dapp/issues/139. Let me know if the image builds on docker hub, so we can finalize this issue or if you encounter the same "file not found" issues as I did. I guess its a context issue.

**Comment by jay-dee7** [comment]
@inspiraluna i was able to build the image using two context paths on an ubuntu machine, one from project root (context path = contrib/docker/dapp) and another by cd'ing into the dir `contrib/docker/dapp`, however i'm not able to see the docker build logs from docker hub, it stays stuck at fetching the logs screen after failing: 
<img width="1315" alt="image" src="https://user-images.githubusercontent.com/10788442/76297833-8614e700-62de-11ea-9cce-7516adb116ec.png">

**Comment by jay-dee7** [comment]
can you share what context did you use? in the docker hub build config

**Comment by jay-dee7** [comment]
this is how my configuration looks like 
<img width="1315" alt="Screenshot 2020-03-10 at 3 02 36 PM" src="https://user-images.githubusercontent.com/10788442/76298941-3e8f5a80-62e0-11ea-9de3-b3c3ad376e7e.png">

**Comment by jay-dee7** [comment]
hey @inspiraluna from my fork, it's building just fine, however the image size magically appears to be 1.7GB which is much much smaller for a first build, can you please pull this image and verify if nothing breaks? also it'll be good if you can give a look at my WIP PR here #140 

docker image: `docker pull jasdeepsingh/doichain-docker:latest`

**Comment by jay-dee7** [comment]
@inspiraluna since the build seems to be working fine, can i raise a claim for the bounty? and also if yes, can i get it in eth?

**Comment by inspiraluna** [comment]
Hey, sure you can claim the bounty. But the payout is only in DOI. Contact me on telegram if you have questions on such questions. also send me your Doichain address in a PM. Alternatively I can send you a paper wallet to your email if you don't have a running node or wallet yet.

**Comment by jay-dee7** [comment]
umm the only reason was because coinbase doesnt have doichain on their list yet and i really wanted to avoid multiple wallets, im not looking into converting it into cryptos as well, just wanna get some and let them ripe. i'm PM'ing you right now on telegram.
